src/main/kotlin/lotto/domain/WinningResult.kt

```diff
@@ -0,0 +1,26 @@
+package lotto.domain
+
+class WinningResult(private val winningStatistics: Map<Int, Int>) {
+    fun calculateTotalPrize(): Int {
+        return winningStatistics.entries.sumOf { (matchCount, count) ->
+            calculatePrizeForMatch(matchCount, count)
```

Comment:
calculatePrizeForMatch 메서드의 로직은 메서드 추출없이 내부 로직을 바로 두더라도 코드가 잘 읽힐 것 같네요. 😃

```suggestion
            val rank = Rank.from(matchCount)
            rank.prize * count
```